pkg/services/searchV2/auth.go

```diff
@@ -0,0 +1,38 @@
+package searchV2
+
+import (
+	"fmt"
+
+	"github.com/grafana/grafana/pkg/models"
+	"github.com/grafana/grafana/pkg/services/sqlstore"
+)
+
+// ResourceFilter checks if a given a uid (resource identifier) check if we have the requested permission
+type ResourceFilter func(uid string) bool
```

Comment:
In my ideal world we would use "path" rather than uid since it encodes both a tree structure and must be unique within an org.  The only advantage of the current `uid` is that the path can change and it stays the same :thinking: we can probably support the same thing using alias or something similar

Reply:
@sdboyer @papagian @radiohead @kylebrandt Are looking at the question of canonicalization of names as part of their current project, fwiw. Maybe they could have some input here.

Reply:
yep, this is a critical thing we need to start providing rules around - search is a great example of a case that benefits from precise semantics here

Reply:
Dare I ask if we can consider something like a file path (in git/s3/gcs/disk) etc a canonical name?